docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/How_Tos/Extend_CMS_Interface.md

```diff
@@ -204,26 +204,443 @@ To make the actions more user-friendly you can also use alternating buttons as
+        # The name of this file will depend on how you've configured your build process
+        - 'my-module/js/dist/main.bundle.js'
+```
+Now that the customisation is applied, our text fields look like this:
+
+![](../../../_images/react-di-1.png)
+
+Let's add another customisation to TextField. If the text goes beyond a specified
+length, let's throw a warning in the UI.
+
+__my-module/js/components/TextLengthChecker.js__
+```js
+const TextLengthCheker = (TextField) => (props) => {  
+  const {limit, value } = props;
+  const invalid = limit !== undefined && value.length > limit;
+
+  return (
+    <div>
+      <TextField {...props} />
+      {invalid &&
+        <span style={{color: 'red'}}>
+          {`Text is too long! Must be ${limit} characters`}
+        </span>
+      }
+    </div>
+  );
+}
+
+export default TextLengthChecker;
+```
+
+We'll apply this one to the injector as well, but let's do it under a different name.
+For the purposes of demonstration, let's imagine this customisation comes from another
+module.
+
+__my-module/js/main.js__
+```js
+import Injector from 'lib/Injector';
+import TextLengthChecker from './components/TextLengthChecker';
+
+Injector.transform('my-other-transformation', (update) => {
+  update('TextField', TextLengthChecker);
+});
+```
+
+Now, both components have applied themselves to the textfield.
+
+![](../../../_images/react-di-2.png)
+
+##### Getting multiple customisations to work together
+
+Both these enhancements are nice, but what would be even better is if they could
+work together collaboratively so that the character count only appeared when the user
+input got within a certain range of the limit. In order to do that, we'll need to be
+sure that the `TextLengthChecker` customisation is loaded ahead of the `CharacterCounter` 
+customisation. 
+
+First let's update the character counter to show characters _remaining_, which is
+much more useful. We'll also update the API to allow a `warningBuffer` prop. This is
+the amount of characters the input can be within the `limit` before the warning shows.
+
+__my-module/js/components/CharacterCounter.js__
+```js
+import React from 'react';
+
+const CharacterCounter = (TextField) => (props) => {
+    const { warningBuffer, limit, value: { length } } = props;
+    const remainingChars = limit - length;
+    const showWarning = length + warningBuffer >= limit;
+    return (
+        <div>
+            <TextField {...props} />
+            {showWarning &&
+            	<small>Characters remaining: {remainingChars}</small>
+            }
+        </div>
+    );
+}
+
+export default CharacterCounter;
+```
+
+Now, when we apply this customisation, we need to be sure it loads _after_ the length
+checker in the middleware chain, as it relies on the prop `limit`. We can do that by specifying priority using `before` and `after`
+metadata to the customisation.
+
+For this example, we'll imagine these two enhancements come from different modules.
```

Comment:
I assume that you're using this to show how you can have multiple modules, as well as interacting enhancements.

I'd probably break this into separate sections, each demonstrating one of these things.